metis-media-service/src/test/java/eu/europeana/metis/mediaprocessing/extraction/PdfToImageConverterTest.java

```diff
@@ -0,0 +1,143 @@
+
+  @BeforeAll
+  static void createMocks() {
+    commandExecutor = mock(CommandExecutor.class);
+    pdfToImageConverter = spy(new PdfToImageConverter(commandExecutor, GHOST_SCRIPT_COMMAND));
+  }
+
+  @BeforeEach
+  void resetMocks() {
+    reset(commandExecutor, pdfToImageConverter);
+  }
+
+  @Test
+  void testDiscoverGhostScriptCommand() throws MediaProcessorException {
+
+    // ghostscript command
+    final String ghostScriptCommand = "gs";
+    final List<String> ghostScriptVersionCommand = Arrays.asList(ghostScriptCommand, "--version");
+
+    // Test right version
+    doReturn("9.26").when(commandExecutor)
+            .execute(eq(ghostScriptVersionCommand), eq(true), any());
+    assertEquals(ghostScriptCommand,
+            PdfToImageConverter.discoverGhostScriptCommand(commandExecutor));
+
+    // Test other commands
+    doReturn("8.26").when(commandExecutor)
+            .execute(eq(ghostScriptVersionCommand), eq(true), any());
+    assertThrows(MediaProcessorException.class,
+            () -> PdfToImageConverter.discoverGhostScriptCommand(commandExecutor));
+    doReturn("Other command").when(commandExecutor)
+            .execute(eq(ghostScriptVersionCommand), eq(true), any());
+    assertThrows(MediaProcessorException.class,
+            () -> PdfToImageConverter.discoverGhostScriptCommand(commandExecutor));
+
+    // Test command execution exception
+    doThrow(new MediaProcessorException("", null)).when(commandExecutor)
+            .execute(eq(ghostScriptVersionCommand), eq(true), any());
+    assertThrows(MediaProcessorException.class,
+            () -> PdfToImageConverter.discoverGhostScriptCommand(commandExecutor));
+  }
+
+  @Test
+  void testCreatePdfConversionCommand() {
+
+    // Create input and output files
+    final Path inputFile = mock(Path.class);
+    doReturn(inputFile).when(inputFile).toAbsolutePath();
+    doReturn("input_file").when(inputFile).toString();
+    final Path outputFile = mock(Path.class);
+    doReturn(outputFile).when(outputFile).toAbsolutePath();
+    doReturn("output_file").when(outputFile).toString();
+
+    // Get the command
+    final List<String> expectedCommand = Arrays.asList(GHOST_SCRIPT_COMMAND,
+            "-q",
+            "-dQUIET",
+            "-dSAFER",
+            "-dBATCH",
+            "-dNOPAUSE",
+            "-dNOPROMPT",
+            "-dMaxBitmap=500000000",
+            "-dAlignToPixels=0",
+            "-dGridFitTT=2",
+            "-sDEVICE=pngalpha",
+            "-dTextAlphaBits=4",
+            "-dGraphicsAlphaBits=4",
+            "-r72x72",
+            "-dFirstPage=1",
+            "-dLastPage=1",
+            "-sOutputFile=" + outputFile.toString(),
+            "-f" + inputFile.toString());
+
+    // Execute the command
+    assertEquals(expectedCommand,
+            pdfToImageConverter.createPdfConversionCommand(inputFile, outputFile));
+  }
+
+  @Test
+  void testConvertToPdf() throws MediaExtractionException {
```

Comment:
Observation:
Why are you grouping test cases in one test method?
Shouldn't we be having a method for each case?

Reply:
That's one approach... personally I don't like to have to repeat the exact same setup multiple times.